app/models/batch_connect/session.rb

```diff
@@ -126,20 +136,34 @@ def save(app:, context:, format: nil)
       self.view       = app.session_view
       self.created_at = Time.now.to_i
 
-      stage(root: app.root.join("template")) &&
+      stage(app.root.join("template"), context: context) &&
         submit(app.submit_opts(context, fmt: format))
     end
 
     # Stage the app's job template to user's staging directory
     # @param root [#to_s] root directory that gets staged
+    # @param context [Object] context available to post-processed staged files
     # @return [Boolean] whether staged successfully
-    def stage(root:)
+    def stage(root, context: nil)
+      # Sync the template files over
       oe, s = Open3.capture2e("rsync -av --delete #{root}/ #{staged_root}")
       unless s.success?
         errors.add(:stage, oe)
         Rails.logger.error(oe)
+        return false
       end
-      s.success?
+
+      # Post process the template files
```

Comment:
Instead of a comment `# Post process template files` maybe more clearly `# Render all template files using ERB`?